src/probnum/filtsmooth/gaussian/approx/_interface.py

```diff
@@ -0,0 +1,119 @@
+"""Temporary interface."""
+
+from typing import Dict, Tuple
+
+from probnum import randprocs, randvars
+
+
+class _LinearizationInterface:
+    """Interface for extended Kalman filtering components."""
+
+    def __init__(
+        self,
+        non_linear_model,
+    ) -> None:
+
+        self.non_linear_model = non_linear_model
+
+        # # Will be constructed later
+        # self.linearized_model = None
+        #
+
+    def forward_realization(
+        self,
+        realization,
+        t,
+        dt=None,
+        compute_gain=False,
+        _diffusion=1.0,
+        _linearise_at=None,
+    ) -> Tuple[randvars.Normal, Dict]:
+        """Approximate forward-propagation of a realization of a random variable."""
+        compute_jacobian_at = (
+            _linearise_at
+            if _linearise_at is not None
+            else randvars.Constant(realization)
+        )
+        linearized_model = self.linearize(t=t, at_this_rv=compute_jacobian_at)
+        return linearized_model.forward_realization(
+            realization=realization,
+            t=t,
+            dt=dt,
+            compute_gain=compute_gain,
+            _diffusion=_diffusion,
+        )
```

Comment:
very soon, this whole class can go, and the linearization is done from the outside in, e.g., an `ExtendedKalman()` object.